include/circt/Dialect/Pipeline/PipelineInterfaces.td

```diff
@@ -0,0 +1,59 @@
+//===- PipelineInterfaces.td - Operation Interfaces --------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This describes the Pipeline operation interfaces.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_Pipeline_PipelineInterfaces
+#define CIRCT_DIALECT_Pipeline_PipelineInterfaces
+
+include "mlir/IR/OpBase.td"
+
+def PipelineLike : OpInterface<"PipelineLike"> {
+  let cppNamespace = "circt::pipeline";
+  let description = "Trait for pipeline-like operations";
+
+  let methods = [
+    InterfaceMethod<"Returns true if the pipeline is latency insensitive",
+      "bool",
+      "isLatencyInsensitive", (ins),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
```

Comment:
I know this is probably just the previous logic factored out, so this may not be relevant to this PR; but, what if some of the inputs/outputs are LI. In other words, what if it's mixed LI non-LI? Would it make sense to return false for both this and `isLatencySensitive` below?

Reply:
It's blurry for me as well. We might consider ripping it out. Add it back later on with `dc.value` interfaces.

Reply:
I'd be in favor of that as well - for a follow-up PR, though!